contrib/platform/src/com/sun/jna/platform/win32/WinGDI.java

```diff
@@ -23,23 +23,481 @@
+        public int dmDisplayFrequency;
+
+        /**
+         * For printers: Specifies how ICM (image color management) is handled. For a non-ICM application, this member determines
+         * if ICM is enabled or disabled. For ICM applications, the system examines this member to determine how to
+         * handle ICM support. This member can be one of the predefined <code>DMICMMETHOD_XXX</code>values, or a driver-defined value
+         * greater than or equal to the value of <code>DMICMMETHOD_USER</code>.
+         *
+         * The printer driver must provide a user interface for setting this member. Most printer drivers support only
+         * the <code>DMICMMETHOD_SYSTEM</code> or <code>DMICMMETHOD_NONE</code> value. Drivers for PostScript printers
+         * support all <code>DCMICMMETHOD_XXX</code>values.
+         *
+         * For displays: This member is not used for displays.
+         */
+        public int dmICMMethod;
+
+        /**
+         * For printers: Specifies which color matching method, or intent, should be used by default. This member is primarily for
+         * non-ICM applications. ICM applications can establish intents by using the ICM functions. This member can be
+         * one of the following predefined values, or a driver defined value greater than or equal to the value of
+         * <code>DMICM_USER</code>.
+         *
+         * For displays: This member is not used for displays.
+         */
+        public int dmICMIntent;
+
+        /**
+         * For printers: Specifies the type of media being printed on. The member can be one of the predefined
+         * <code>DMMEDIA_XXX</code> values, or a driver-defined value greater than or equal to the value of <code>DMMEDIA_USER</code>.
+         *
+         * To retrieve a list of the available media types for a printer, use the <code>DeviceCapabilities</code>
+         * function with the <code>DC_MEDIATYPES</code> flag.
+         *
+         * For displays: This member is not used for displays.
+         */
+        public int dmMediaType;
+
+        /**
+         * For printers: Specifies how dithering is to be done. The member can be one of the predefined
+         * <code>DMDITHER_XXX</code> values, or a driver-defined value greater than or equal to the value of
+         * <code>DMDITHER_USER</code>.
+         *
+         * For displays: This member is not used for displays.
+         */
+        public int dmDitherType;
+
+        /**
+         * Not used; must be zero.
+         */
+        public int dmReserved1;
+
+        /**
+         * Not used; must be zero.
+         */
+        public int dmReserved2;
+
+        /**
+         * This member must be zero.
+         */
+        public int dmPanningWidth;
+
+        /**
+         * This member must be zero.
+         */
+        public int dmPanningHeight;
+
+        /**
+         * Converts dmDeviceName from raw byte[] to String
+         */
+        public String getDmDeviceName() {
```

Comment:
I am making the assumption that the two name fields are null terminated. I have seen data overwritten in the fields, and the old name was not completely removed, just written over with a null terminated string. Windows also limits printer name to 31 chars, for a 32 char long buffer.